src/sage/categories/finite_dimensional_lie_algebras_with_basis.py

```diff
@@ -1594,6 +1594,165 @@ def universal_commutative_algebra(self):
             R = P[0].parent()
             return R.quotient(P)
 
+        def casimir_element(self, order=2, UEA=None, force_generic=False):
+            r"""
+            Return the Casimir element in the universal enveloping algebra
+            of ``self``.
+
+            A *Casimir element* of order `k` is a distinguished basis element
+            for the center of `U(\mathfrak{g})` of homogeneous degree `k`
+            (that is, it is an element of `U_k \setminus U_{k-1}`, where
+            `\{U_i\}_{i=0}^{\infty}` is the natural filtration of
+            `U(\mathfrak{g})`. When `\mathfrak{g}` is a simple Lie algebra,
+            then this spans `Z(U(\mathfrak{g}))_k`.
+
+            INPUT:
+
+            - ``order`` -- (default: ``2``) the order of the Casimir element
+            - ``UEA`` -- (optional) the universal enveloping algebra to
+              return the result in
+            - ``force_generic`` -- (default: ``False``) if ``True`` for the
+              quadradic order, then this uses the default algorithm; otherwise
+              this is ignored
+
+            ALGORITHM:
+
+            For the quadradic order (i.e., ``order=2``), then this uses
+            K^{ij}`, the inverse of the Killing form matrix, to compute
+            `C_{(2)} = \sum_{i,j} K^{ij} X_i \cdots X_j`, where `\{X_1, \ldots,
+            X_n\}` is a basis for `\mathfrak{g}`. Otherwise this solves the
+            system of equations
+
+            .. MATH::
+
+                f_{aj}^b \kappa^{jc\cdots d} + f_{aj}^c \kappa^{cj\cdots d}
+                \cdots + f_{aj}^d \kappa^{bc \cdots j}
+
+            for the symmetric tensor `\kappa^{i_1 \cdots i_k}`, where `k`
+            is the ``order``. This system comes from `[X_i, C_{(k)}] = 0`
+            with `C_{(k)} = \sum_{i_1, \ldots, i_k\}^n
+            \kappa^{i_1 \cdots i_k} X_{i_1} \cdots X_{i_k}`.
+
+            EXAMPLES::
+
+                sage: L = LieAlgebra(QQ, cartan_type=['A', 1])
+                sage: C = L.casimir_element(); C
+                1/8*b1^2 + 1/2*b0*b2 - 1/4*b1
+                sage: U = L.universal_enveloping_algebra()
+                sage: all(g * C == C * g for g in U.gens())
+                True
+                sage: U = L.pbw_basis()
+                sage: C = L.casimir_element(UEA=U); C
+                1/2*PBW[alpha[1]]*PBW[-alpha[1]] + 1/8*PBW[alphacheck[1]]^2
+                 - 1/4*PBW[alphacheck[1]]
+                sage: all(g * C == C * g for g in U.algebra_generators())
+                True
+
+                sage: L = LieAlgebra(QQ, cartan_type=['B', 2])
+                sage: U = L.pbw_basis()
+                sage: C = L.casimir_element(UEA=U)
+                sage: all(g * C == C * g for g in U.algebra_generators())
```

Comment:
do we have a test function `is_central`  ?

Reply:
Not that I am aware at this level of generality (there is one for semigroup algebras though). It certainly wouldn’t be difficult to add it to the appropriate categories. Although I am slightly worried about ambiguities with “central as an object in category X” and compared to supecenters. Everything is probably okay, but it would need its own PR.

Reply:
yes, sure